CSS :state() pseudo class (#32000)
mdn/content: files/en-us/web/api/web_components/using_custom_elements/index.md

customElements.define("my-custom-element", MyCustomElement);
```

> After adding `<my-custom-element>` to the HTML we can use the identifier added to the `CustomStateSet`, passed to the [`:state()`](/en-US/docs/Web/CSS/:state) function, as a custom state pseudo-class for selecting the element state.

Reviewer: ```suggestion
After adding `<my-custom-element>` to the HTML, we can use the identifier added to `CustomStateSet` and pass it to the [`:state()`](/en-US/docs/Web/CSS/:state) pseudo-class for matching the custom element's custom state.
```

Author: Tweaked

```suggestion
We can use the identifier added to the custom element's `CustomStateSet` (`this._internals.states`) for matching the element's custom state.
This is matched by passing the identifier to the [`:state()`](/en-US/docs/Web/CSS/:state) pseudo-class.
```

}
```

> The `:state()` pseudo class can also be used within the [`:host()`](/en-US/docs/Web/CSS/:host_function) CSS pseudo-class function to match a custom state [within custom element's shadow DOM](/en-US/docs/Web/CSS/:state#matching_a_custom_state_in_a_custom_elements_shadow_dom), or after the [`::part()`](/en-US/docs/Web/CSS/::part) pseudo element to match the [shadow parts](/en-US/docs/Web/CSS/CSS_shadow_parts) of a custom element that have a particular state.

Reviewer: It is a long sentence, splitting for readability
```suggestion
The `:state()` pseudo-class can also be used within the [`:host()`](/en-US/docs/Web/CSS/:host_function) pseudo-class function to match a custom state [within a custom element's shadow DOM](/en-US/docs/Web/CSS/:state#matching_a_custom_state_in_a_custom_elements_shadow_dom). Additionally, the `:state()` pseudo-class can be used after the [`::part()`](/en-US/docs/Web/CSS/::part) pseudo-element to match the [shadow parts](/en-US/docs/Web/CSS/CSS_shadow_parts) of a custom element that is in a particular state.
```

> There several live examples in {{domxref("CustomStateSet")}} showing how this works.

Reviewer: ```suggestion
There are several live examples in {{domxref("CustomStateSet")}} showing how this works.
```
